[Footer] Brand information block

LucasLacerdaUX opened the pull request:
### PR Summary: 
We added a new Brand Information block to the Footer section. The content can be configured on Theme Settings.

### Why are these changes introduced?
Introduces a new Brand Information block to the Footer.
Fixes #2092.

### What approach did you take?
- The `Brand Information` content lives in Theme Settings. This follows a similar pattern to our Social Media icons, and we intentionally designed it this way (future project requirement).
- Re-used classes from other footer blocks
- Moved Social Media icons into a `social-icons` snippet.
- Social Media icons and settings have been re-ordered to follow popularity of each platform
- We've increased the maximum size in the `Image` block. 


### Testing steps/scenarios
- [ ] Add a new Brand Information block to the Footer
- [ ] If the Brand Information is the sole block on the Footer, content should be centered. Otherwise, left-aligned.
- [ ] Add Social Media
- [ ] Tweak image size

### Demo links
- [Store](https://os2-demo.myshopify.com/?preview_theme_id=139158257686)
- [Editor](https://os2-demo.myshopify.com/admin/themes/139158257686/editor)

### Checklist
- [x] Added PR summary for [release notes](https://themes.shopify.com/themes/dawn/styles/default#ReleaseNotes)
- [x] Requested review from UX (Only for changes that are affecting the experience or perceivable visual details)
- [ ] Created a ticket for the [help.shopify.com](https://help.shopify.com) documentation team about updates to theme settings. (Internal-only task)
- [x] Followed [theme code principles](https://github.com/Shopify/dawn/blob/main/.github/CONTRIBUTING.md#theme-code-principles)
- [x] Linted with [Theme Check](https://github.com/Shopify/theme-check)
- [x] Tested on [mobile](https://shopify.dev/themes/store/requirements#mobile-browser-requirements)
- [x] Tested on [multiple browsers](https://shopify.dev/themes/store/requirements#desktop-browser-requirements)
- [x] Tested for [accessibility](https://shopify.dev/themes/best-practices/accessibility)

allylane commented:
<img width="1085" alt="Screenshot 2022-12-02 at 2 45 44 PM" src="https://user-images.githubusercontent.com/17570703/205374003-ef18ad2f-ae12-46ed-a14d-cd356e60b6f7.png">
If there's no content in this section, it shouldn't have padding.

allylane commented:
<img width="1120" alt="Screenshot 2022-12-02 at 2 53 20 PM" src="https://user-images.githubusercontent.com/17570703/205375010-de9f4e2b-85dd-4471-974e-45b6bcf6c984.png">
Ideally we'll have a border top on this section. 
We could add a checkbox to "Footer" to show or hide it so it's not on every theme.

allylane commented:
As we chatted about, let's put the footer links on a new line.
<img width="1069" alt="Screenshot 2022-12-02 at 3 05 52 PM" src="https://user-images.githubusercontent.com/17570703/205376788-2a42c01c-e925-4843-be1e-e772ab48d952.png">

ludoboludo commented on sections/footer.liquid:
```diff
@@ -73,14 +73,36 @@
-                        src="{{ block.settings.image | image_url: width: 400 }}"
+                        src="{{ block.settings.image | image_url: width: 760 }}"
                         alt="{{ block.settings.image.alt | escape }}"
+                        class="global-media-settings"
```
Pseudo elements don't work on `image` elements. So only the border radius and border work. Is that expected or should the shadow be ideally coming up ?
<details>
<summary>Screenshot</summary>

[docs](https://developer.mozilla.org/en-US/docs/Web/CSS/::after)
![](https://screenshot.click/02-19-w91qb-475m7.png)
</details>

ludoboludo commented on sections/footer.liquid:
```diff
@@ -73,14 +73,36 @@
+                {%- when 'brand_information' -%}
+                  <div class="footer-block-brand-information">
+                    {%- if settings.brand_image != blank -%}
+                      {%- assign image_size_2x = settings.brand_image.width | times: 2 | at_most: 5760 -%}
```
I think here, since we know what the max value can be, `550px` then we should use that. Otherwise if a merchant use a large image we're going to be loading an image super heavy without any real benefit but a hit on performance. 
<details>
<summary>Screenshot</summary>

![](https://screenshot.click/02-26-r7guo-0nluw.png)
</details>

And when it uses `at_most: 550` then the weight is 10 times lighter and still look good 👍

ludoboludo commented on sections/footer.liquid:
```diff
@@ -73,14 +73,36 @@
+                      <img
+                        srcset= "{{ settings.brand_image | image_url: width: settings.brand_image.width }}, {{ settings.brand_image | image_url: width: image_size_2x }} 2x"
+                        src="{{ settings.brand_image | image_url: width: 760 }}"
+                        alt="{{ settings.brand_image.alt | escape }}"
```
I've noticed in another PR that we tend to use the `| escape` filter often for alt text but I'm not sure why 🤔

ludoboludo commented on sections/footer.liquid:
```diff
@@ -73,14 +73,36 @@
+                  <div class="footer-block-brand-information">
+                    {%- if settings.brand_image != blank -%}
+                      {%- assign image_size_2x = settings.brand_image.width | times: 2 | at_most: 5760 -%}
+                      <img
+                        srcset= "{{ settings.brand_image | image_url: width: settings.brand_image.width }}, {{ settings.brand_image | image_url: width: image_size_2x }} 2x"
+                        src="{{ settings.brand_image | image_url: width: 760 }}"
+                        alt="{{ settings.brand_image.alt | escape }}"
+                        class="global-media-settings"
+                        loading="lazy"
+                        width="{{ settings.brand_image.width }}"
+                        height="{{ settings.brand_image.height }}"
+                        style="max-width: min(100%, {{ settings.brand_image_width }}px);"
+                      >
```
should we leverage the image_tag filter instead ?

LucasLacerdaUX replied:
Updated to use `image_tag`

ludoboludo commented on sections/footer.liquid:
```diff
@@ -73,14 +73,36 @@
+                    {%- if settings.brand_image != blank -%}
+                      {%- assign image_size_2x = settings.brand_image.width | times: 2 | at_most: 5760 -%}
+                      <img
+                        srcset= "{{ settings.brand_image | image_url: width: settings.brand_image.width }}, {{ settings.brand_image | image_url: width: image_size_2x }} 2x"
```
```suggestion
                        srcset= "{{ settings.brand_image | image_url: width: settings.brand_image_width }}, {{ settings.brand_image | image_url: width: image_size_2x }} 2x"
```

ludoboludo commented on sections/footer.liquid:
```diff
@@ -73,14 +73,36 @@
+                        style="max-width: min(100%, {{ settings.brand_image_width }}px);"
+                      >
+                    {%- endif -%}
+                    <h2 class="footer-block__heading">{{ settings.brand_headline }}</h2>
+                    <p>{{ settings.brand_description }}</p>
```
We ok with rendering those elements when not content is added in the settings for them ?

LucasLacerdaUX replied:
Updated to add a check for emptiness.

ludoboludo commented on sections/footer.liquid:
```diff
@@ -73,14 +73,36 @@
                 {%- when 'image' -%}
                   <div class="footer-block__details-content footer-block-image {{ block.settings.alignment }}">
                     {%- if block.settings.image != blank -%}
                       {%- assign image_size_2x = block.settings.image_width | times: 2 | at_most: 5760 -%}
```
I guess my comment would apply here though it's been like that for a while.

ludoboludo commented on config/settings_schema.json:
```diff
@@ -33,6 +33,38 @@
+      {
+        "type": "richtext",
+        "id": "brand_headline",
+        "default": "<p></p>",
```
I wonder if we should have something here 🤔 I guess we don't want to create work for people to go and delete it. 
<details>
<summary>Screenshot</summary>

![](https://screenshot.click/02-40-3x4ru-bp5a6.png)
</details>

metamoni replied:
We shouldn't use `<p></p>` as the default value here, as this results in invalid HTML. A `p` element can't be a child of a heading.
![invalid](https://user-images.githubusercontent.com/22390758/205589654-2d5f65e4-858a-48ee-b731-75dd65a40074.png)

ludoboludo commented on assets/section-footer.css:
```diff
@@ -479,11 +479,30 @@ noscript .localization-selector.link {
+
 .footer-block__details-content .placeholder-svg {
   max-width: 20rem;
 }
```
Not related to any of your changes on the PR but the default column gap value looks tight on those footer columns: 
<details>
<summary>Screenshot</summary>

![](https://screenshot.click/02-51-q498i-puil5.png)
</details>

LucasLacerdaUX commented on locales/en.default.schema.json:
```diff
@@ -904,6 +921,21 @@
+            },
+            "show_social": {
+              "label": "Show social media icons",
+              "info": "To display your social media accounts, link them in your theme settings."
```
Good catch. I've updated the translation files to link to the social media tab.

metamoni commented on config/settings_schema.json:
```diff
@@ -33,6 +33,38 @@
+      {
+        "type": "richtext",
+        "id": "brand_description",
+        "default": "<p></p>",
```
Hmm, I'm still seeing this on the page though 🤔
![Screenshot 2022-12-08 at 09 38 04](https://user-images.githubusercontent.com/22390758/206411824-76a16de5-54f1-4097-ba75-5590ba6f668c.png)

metamoni replied:
I'm still seeing it in the header, and the logo alignment seems to have changed as well

https://user-images.githubusercontent.com/22390758/206412866-461ddf77-c1fc-49d7-9d85-284eae948c9e.mp4

LucasLacerdaUX replied:
@metamoni That was a sync error on the demo link, the type was changed to `inline_richtext` which does not have wrapping `<p>` tags on [this commit](https://github.com/Shopify/dawn/pull/2112/commits/aee2a28a7a349ce8a43e5a5b3e0a0a3cadc186cc).

<img width="813" alt="image" src="https://user-images.githubusercontent.com/37168033/206794881-c9577c49-ebfb-4a8e-bedd-3515519310b6.png">

I just re-connected the theme and it should now be gone. Let me know if you find any additional issues. I'll continue addressing the other feedback :D

LucasLacerdaUX commented:
@allylane 

> If there's no content in this section, it shouldn't have padding.

That's a good catch. This will require additional work as it's something that is already present with current blocks: if you add an empty image block, it'll have padding. I'll try to address this in a separate PR.

> When I upload a logo, the brand information section gets cut off.

That's a Theme Editor issue and not related to the theme. I can flag this to the team. Thanks.

> As we chatted about, let's put the footer links on a new line.

Already done in a separate PR (#2084).

> Everything in the design is a bit smaller. Can we bring the size down and bit, and increase the opacity of the body copy:

I'm using the values in our design system. This ensures consistency with other blocks such as the `Text block`:

<img width="1160" alt="image" src="https://user-images.githubusercontent.com/37168033/207064160-25ba549b-4124-4454-b50d-f0cb98e02e18.png">

I would suggest to not change opacity of the body copy - or at least not as part of this PR without proper review of everything else - it's a global property on the theme and may affect a lot more than we expect to.

The text size is also following the defaults in the rest of the theme. Are we sure we want to make a special case for this block that will look different from everything else?

As for the social icons, I can easily update their size to be slightly smaller. Would it be okay to change it for both `Brand info` and `Footer`, though?

LucasLacerdaUX commented:
Updated PR to:
- Fix spacing between image and Headline
- Updated to support RTE link styling
- Updated content copyright font size to 11px instead of 10px.

ludoboludo commented on assets/section-footer.css:
```diff
@@ -463,16 +464,54 @@ noscript .localization-selector.link {
+    var(--media-shadow-vertical-offset)
+    var(--media-shadow-blur-radius)
+    rgba(var(--color-shadow), var(--media-shadow-opacity));
+  margin-bottom: calc(2rem + var(--media-shadow-vertical-offset) * var(--media-shadow-visible));
```
this doesn't work well when the vertical offset is a negative value. 
I think on the product page we did it using `padding` so that when it's a negative value, nothing happens 🤔 
But we could also use a `max()`. `max((var(--media-shadow-vertical-offset) * var(--media-shadow-visible)), 0rem)`

ludoboludo commented on assets/section-footer.css:
```diff
@@ -463,16 +464,54 @@ noscript .localization-selector.link {
 }
 
+.footer-block:only-child .footer-block-brand-information .footer__list-social.list-social {
+  justify-content: center;
```
I think here we would need to also remove the negative margin left to make sure it's properly centered. Or add a negative `margin-right`
[video](https://screenshot.click/09-30-64en6-l8g6r.mp4)

LucasLacerdaUX replied:
Updated with `margin-right`. Removing `margin-left` would break the layout when icons are left aligned :D

ludoboludo commented on sections/footer.liquid:
```diff
@@ -73,14 +73,43 @@
                     </ul>
                   {%- endif -%}
+                {%- when 'brand_information' -%}
+                  <div class="footer-block-brand-information">
```
Nitpick to match the other blocks.
```suggestion
                  <div class="footer-block__brand-information">
```

ludoboludo commented on sections/footer.liquid:
```diff
@@ -73,14 +73,43 @@
+                          max-width: min(100%, {{ settings.brand_image_width }}px);
+                        }
+                      </style>
+                      <div class="brand-information__image-wrapper global-media-settings">
```
This class doesn't have a parent `brand-information` one to justify the `__image-wrapper` so maybe it could just be `brand-information-image-wrapper` ? 🤷

ludoboludo commented on sections/footer.liquid:
```diff
@@ -73,14 +73,43 @@
-                        src="{{ block.settings.image | image_url: width: 400 }}"
+                        src="{{ block.settings.image | image_url: width: 760 }}"
                         alt="{{ block.settings.image.alt | escape }}"
+                        class="global-media-settings"
```
This image would also need a wrapping div with that class like you did for the branding block. It might have got lost in the rebases.
<details>
<summary>Screenshot</summary>

![](https://screenshot.click/09-46-03nig-ij2bj.png)
</details>

LucasLacerdaUX replied:
Updated :D

LucasLacerdaUX commented:
@ludoboludo @metamoni Addressed your comments! Let me know if I missed anything! :D

metamoni commented:
@LucasLacerdaUX when you remove both the Quick links menu and the email signup form there's a lot of whitespace at the bottom of the Brand info block compared to the top. Should these be equal?

Also not seeing the social media icons, for some reason, even though they're enabled. Maybe rebasing the branch might fix it?
![spacing](https://user-images.githubusercontent.com/22390758/211847773-06940529-df98-40b1-8d1e-ee8a5ccd23bd.png)

LucasLacerdaUX commented:
@metamoni The social icons are working fine for me on the demo store in the PR description. Did you add the social media to the Theme Settings?

I've rebased the branch to the latest main - with no conflicts.

As for the additional spacing, this is already in main and happening to other blocks like Menu. I'll tag @allylane to check if this is intentional and maybe we can fix in the follow up PR.
<img width="443" alt="image" src="https://user-images.githubusercontent.com/37168033/211874397-2c109960-cf78-4058-9d39-d2e8746c3318.png">

ludoboludo commented on assets/section-footer.css:
```diff
@@ -457,22 +458,80 @@ noscript .localization-selector.link {
+
+.footer-block-image.left {
+  justify-content: flex-start;
+}
+
+.footer-block-image.center {
+  justify-content: center;
+}
+
+.footer-block-image.right {
+  justify-content: flex-end;
+}
+
 @media screen and (max-width: 749px) {
-  .footer-block-image {
-    text-align: center;
+  .footer-block-image,
+  .footer-block-image.left,
+  .footer-block-image.center,
+  .footer-block-image.right {
+    justify-content: center;
   }
 }
```
This seem a bit opposite to our usual approach where we tend to have default values on mobile and overwrites in media queries for tablet/desktop, no ?

LucasLacerdaUX replied:
Good point. That's how it was originally designed so I didn't change, but I can refactor this in the follow up PR 👀

ludoboludo commented on sections/footer.liquid:
```diff
@@ -75,19 +75,43 @@
                       {%- endfor -%}
                     </ul>
                   {%- endif -%}
+                {%- when 'brand_information' -%}
```
I wonder if we should/could re organize the general settings to match the output order: 
<details>
<summary>Screenshot</summary>

![](https://screenshot.click/11-14-srabo-f4nlg.png)
</details>

Seems like brand info is only used in the footer so far 🤔 So it should be ok

LucasLacerdaUX replied:
cc @allylane

ludoboludo reviewed (approved):
Looking good. Added some little nitpicks but nothing blocking 👍

LucasLacerdaUX commented on sections/footer.liquid:
```diff
@@ -75,19 +75,43 @@
+                      <h2 class="footer-block__heading rte">{{ settings.brand_headline }}</h2>
+                    {%- endif -%}
+                    {%- if settings.brand_description != blank -%}
+                      <div class="rte">{{ settings.brand_description }}</div>
```
Will also include in the follow up PR :D
